### PR TITLE
GOLD-233: Implement method whitelisting middleware to secure API endpoints

### PR DESCRIPTION
**Pull request by S0naliThakur**
Linear: https://linear.app/shm/issue/GOLD-233
Summary: Added middleware to restrict API access to only whitelisted methods.

### CODE REVIEW

File: src/middlewares/methodWhitelist.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Request, Response, NextFunction } from 'express';
+import { methods } from '../api';
+
+const allowedMethods = Object.keys(methods);
+
+export const methodWhitelist = (req: Request, res: Response, next: NextFunction) => {
+  const method = req.body?.method;
```
Comment (arhamj) [inline src/middlewares/methodWhitelist.ts:7]: If there is no method we are falling back to the next(). Maybe we can invert the check to call next and if not fail with 403.

**Review by arhamj — CHANGES_REQUESTED** [review]
Minor changes requested. Mostly LGTM.

File: src/middlewares/methodWhitelist.ts (hunk 1 above)
Comment (S0naliThakur) [reply src/middlewares/methodWhitelist.ts:7]: updated!

**Review by timadinorth — APPROVED** [review]

**Review by kun6fup4nd4 — DISMISSED** [review]

File: src/middlewares/methodWhitelist.ts (hunk 1 above)
Comment (timadinorth) [reply src/middlewares/methodWhitelist.ts:7]: original goal - if method is in whitelist we continue with next, if not, return error, correct ?

**Pull request merged**